### PR TITLE
190-strange-coloring-ega-mode

### PR DESCRIPTION
**Pull request by jinek**

### CODE REVIEW

File: src/Tests/Consolonia.Core.Tests/WithLifetimeFixture/LifetimeSetupFixture.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+using Consolonia.Core.Dummy;
+using NUnit.Framework;
+using static System.GC;
+
+namespace Consolonia.Core.Tests.WithLifetimeFixture
+{
+    [SetUpFixture]
+    public class LifetimeSetupFixture : IDisposable
+    {
+        private class ContextApp2 : Application;
```
[the remark this answers is not in the document]
Comment (jinek) [reply src/Tests/Consolonia.Core.Tests/WithLifetimeFixture/LifetimeSetupFixture.cs:16]: learn c#

**Review by tomlm — DISMISSED** [review]

**Comment by tomlm** [comment]
Oh shoot, I moved line brush. Sorry. Arghhh

File: src/Consolonia.PlatformSupport/PlatformSupportExtensions.cs [hunk 2]
```diff
@@ -25,7 +27,36 @@ public static AppBuilder UseAutoDetectedConsole(this AppBuilder builder)
...
+                            "xterm-256color" or "screen-256color" or "tmux-256color" => new RgbConsoleColorMode(),
+                            _ => new EgaConsoleColorMode()
+                        };
+                        break;
+                    default:
+                        result = new EgaConsoleColorMode();
+                        break;
+                }
+
+            return builder.UseConsoleColorMode(result);
```
[the remark this answers is not in the document]
Comment (jinek) [reply src/Consolonia.PlatformSupport/PlatformSupportExtensions.cs:59]: default is a fallback!

File: src/Consolonia.Core/Drawing/PixelBufferImplementation/EgaConsoleColor/EgaConsoleColorMode.cs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                    ConsoleColor.Gray => 7,
+                    ConsoleColor.DarkGray => 8,
+                    ConsoleColor.Red => 9,
+                    ConsoleColor.Green => 10,
+                    ConsoleColor.Yellow => 11,
+                    ConsoleColor.Blue => 12,
+                    ConsoleColor.Magenta => 13,
+                    ConsoleColor.Cyan => 14,
+                    ConsoleColor.White => 15,
+                    _ => 7 // Default to white if unknown
```
Comment (tomlm) [inline src/Consolonia.Core/Drawing/PixelBufferImplementation/EgaConsoleColor/EgaConsoleColorMode.cs:99]: is it possible to detect default color when app starts?  Maybe future enhancement.

File: src/Consolonia.Core/Drawing/PixelBufferImplementation/EgaConsoleColor/EgaConsoleColorMode.cs [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+                };
+
+                return ansiCode < 8
+                    ?
+                    // Standard colors
+                    $"\x1b[{(isBackground ? 40 + ansiCode : 30 + ansiCode)}m"
+                    :
+                    // Bright colors
+                    $"\x1b[{(isBackground ? 100 + (ansiCode - 8) : 90 + (ansiCode - 8))}m";
```
Comment (tomlm) [inline src/Consolonia.Core/Drawing/PixelBufferImplementation/EgaConsoleColor/EgaConsoleColorMode.cs:108]: can we add to Esc.cs ?

File: src/Consolonia.Core/Drawing/PixelBufferImplementation/Pixel.cs [hunk 5]
```diff
@@ -149,29 +150,8 @@ public Pixel Blend(Pixel pixelAbove)
...
-
-            // Convert back to [0..255]
-            byte a = (byte)(outAlpha * 255f);
-            byte r = (byte)(outR * 255f);
-            byte g = (byte)(outG * 255f);
-            byte b = (byte)(outB * 255f);
-
-            return Color.FromArgb(a, r, g, b);
+            var consoleColorMode = AvaloniaLocator.Current.GetRequiredService<IConsoleColorMode>();
+            return consoleColorMode.Blend(target, source);
```
Comment (tomlm) [inline src/Consolonia.Core/Drawing/PixelBufferImplementation/Pixel.cs:154]: ooh, sliick

File: src/Consolonia.Core/Drawing/PixelBufferImplementation/PixelBuffer.cs [hunk 6]
```diff
@@ -26,7 +27,7 @@ public PixelBuffer(ushort width, ushort height)
             // blended into it.
             for (ushort y = 0; y < height; y++)
             for (ushort x = 0; x < width; x++)
-                _buffer[x, y] = Pixel.Space;
+                _buffer[x, y] = new Pixel(new PixelBackground(Colors.Black));
```
Comment (tomlm) [inline src/Consolonia.Core/Drawing/PixelBufferImplementation/PixelBuffer.cs:30]: Why not transparent background?

**Review by tomlm — APPROVED** [review]

**Pull request merged**